code/obj/item/clothing/injector_mask_belt.dm

```diff
@@ -427,3 +406,153 @@ There's A LOT of duplicate code here, which isn't ideal to say the least. Should
+			.["conditionTreshold"]  = list(
+				currentValue = tresholdCondition.threshold,
+				suffix = tresholdCondition.suffix,
+				minValue = tresholdCondition.minValue,
+				maxValue = tresholdCondition.maxValue,
+			)
+
+			if (istype(condition, /datum/injector_belt_condition/with_threshold/damage))
+				var/datum/injector_belt_condition/with_threshold/damage/damageCondition = condition
+				.["conditionDamage"]  = list(
+					damagetype = damageCondition.damagetype
+				)
+
+	.["injectionAmount"] = inj_amount
+	.["minimumTime"] = min_time
+
+/proc/autoinjector_ui_act(obj/source, action, params, mob/user, var/maximum_volume)
+	var/obj/item/injector_belt/current_belt = null
+	var/obj/item/clothing/mask/gas/injector_mask/current_mask = null
+	if (istype(source, /obj/item/injector_belt))
+		current_belt = source
+	else if (istype(source, /obj/item/clothing/mask/gas/injector_mask))
+		current_mask = source
+
+	switch(action)
+		if ("remove_cont")
+			if (current_belt)
+				current_belt.container.set_loc(get_turf(source))
+				current_belt.container = null
+			else if (current_mask)
+				current_mask.container.set_loc(get_turf(source))
+				current_mask.container = null
+			. = TRUE
+
+		if ("remove_cond")
+			if (current_belt)
+				current_belt.condition = null
+			else if (current_mask)
+				current_mask.condition = null
+			. = TRUE
+
+		if ("sel_cond")
+			var/list/filtered = autoinjector_trigger_names(TRUE)
+
+			var/selected = params["condition"]
+			var/selected_type
+			var/datum/injector_belt_condition/selected_condition = null
+			if (selected <> "None")
+				selected_type = filtered[selected]
+				selected_condition = new selected_type()
+
+			if (current_belt)
+				current_belt.condition = selected_condition
+			else if (current_mask)
+				current_mask.condition = selected_condition
+
+			if (selected_condition && !istype(selected_condition, /datum/injector_belt_condition/with_threshold))
+				selected_condition.setup(user)
+			. = TRUE
+
+		if ("sel_damage_type")
+			var/datum/injector_belt_condition/with_threshold/damage/currentCondition
+			if (current_belt)
+				currentCondition = current_belt.condition
+			else if (current_mask)
+				currentCondition = current_mask.condition
+
+			currentCondition.setupDamageType(user, params["damagetype"])
+			. = TRUE
+
+		if("changeConditionValue")
+			var/datum/injector_belt_condition/currentCondition
+			if (current_belt)
+				currentCondition = current_belt.condition
+			else if (current_mask)
+				currentCondition = current_mask.condition
+
+			currentCondition.setup(user, params["conditionValue"])
+			. = TRUE
+
+		if("changeAmount")
+			var/clampedVolume = clamp(round(params["amount"]), 1, maximum_volume);
+			if (current_belt)
+				current_belt.inj_amount = clampedVolume
+			else if (current_mask)
+				current_mask.inj_amount = clampedVolume
+			. = TRUE
+
+		if ("changeMintime")
+			var/clampedTime = clamp(round(params["mintime"]), 3, 300)
```

Comment:
3 whats? `SECONDS`?
should delete the `*10` in the `SPAWN(min_time*10) can_trigger = 1` line in this file and add units to these